api/models/user.go

```diff
@@ -0,0 +1,62 @@
+package models
+
+import (
+	"context"
+	"fmt"
+	"unicode"
+
+	"github.com/PedroChaparro/loomies-backend/configuration"
+	"github.com/PedroChaparro/loomies-backend/interfaces"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var Collection *mongo.Collection = configuration.ConnectToMongoCollection("users")
+var User interfaces.User
+
+func CheckExistEmail(email string) error {
+
+	err := Collection.FindOne(
+		context.TODO(),
+		bson.D{{"email", email}},
+	).Decode(&User)
+
+	return err
+
+}
+
+func CheckExistUser(user string) error {
+
+	err := Collection.FindOne(
+		context.TODO(),
+		bson.D{{"user", user}},
+	).Decode(&User)
+
+	return err
+
+}
+
+func InsertUser(data map[string]interface{}) error {
```

Comment:
I think this function can receive an `interfaces.User` instead of `map[string]interface{}` which is less strict.